### PR TITLE
[SPARK-45729][PYTHON][DOCS] Fix PySpark testing guide links

### PR DESCRIPTION
**Pull request by asl3**
### What changes were proposed in this pull request?
This PR fixes links in the PySpark testing guidelines page (hyperlinks words instead of displaying the URLs).

### Why are the changes needed?
The official PySpark testing guidelines page is the highest-ranking Google Search result for PySpark testing. These changes clean up the guide and improve the user experience.

### Does this PR introduce _any_ user-facing change?
Yes, the changes affect the user-facing testing guidelines page.

### How was this patch tested?
Existing tests

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/docs/source/getting_started/testing_pyspark.ipynb [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "\n",
     "This guide is a reference for writing robust tests for PySpark code.\n",
     "\n",
-    "To view the docs for PySpark test utils, see here. To see the code for PySpark built-in test utils, check out the Spark repository here. To see the JIRA board tickets for the PySpark test framework, see here."
+    "To view the docs for PySpark test utils, see here. To see the code for PySpark built-in test utils, check out the Spark repository here. To see the JIRA board tickets for the PySpark test framework, see [here](https://issues.apache.org/jira/browse/SPARK-44042)."
```
Comment (allisonwang-db) [inline python/docs/source/getting_started/testing_pyspark.ipynb:12]: I think we can just mention the testing utility functions' API doc, and remove the rest of the external links
```suggestion
    "To view the docs for PySpark test utils, see [here](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.testing.assertDataFrameEqual.html. "
```

File: python/docs/source/getting_started/testing_pyspark.ipynb [hunk 2]
```diff
@@ -193,7 +193,7 @@
     "### Option 2: Using [Unit Test](https://docs.python.org/3/library/unittest.html)\n",
     "For more complex testing scenarios, you may want to use a testing framework.\n",
     "\n",
-    "One of the most popular testing framework options is unit tests. Let’s walk through how you can use the built-in Python `unittest` library to write PySpark tests. For more information about the `unittest` library, see here: https://docs.python.org/3/library/unittest.html.  \n",
+    "One of the most popular testing framework options is unit tests. Let’s walk through how you can use the built-in Python `unittest` library to write PySpark tests. For more information about the `unittest` library, see [here](https://docs.python.org/3/library/unittest.html).  \n",
```
Comment (allisonwang-db) [inline python/docs/source/getting_started/testing_pyspark.ipynb:196]:
```suggestion
    "One of the most popular testing framework options is unit tests. Let’s walk through how you can use the built-in Python `unittest` library to write PySpark tests. \n",
```

File: python/docs/source/getting_started/testing_pyspark.ipynb [hunk 3]
```diff
@@ -273,7 +273,7 @@
    "source": [
     "### Option 3: Using [Pytest](https://docs.pytest.org/en/7.1.x/contents.html)\n",
     "\n",
-    "We can also write our tests with `pytest`, which is one of the most popular Python testing frameworks. For more information about `pytest`, see the docs here: https://docs.pytest.org/en/7.1.x/contents.html.\n",
+    "We can also write our tests with `pytest`, which is one of the most popular Python testing frameworks. For more information about `pytest`, see the docs [here](https://docs.pytest.org/en/7.1.x/contents.html).\n",
```
Comment (allisonwang-db) [inline python/docs/source/getting_started/testing_pyspark.ipynb:276]:
```suggestion
    "We can also write our tests with `pytest`, which is one of the most popular Python testing frameworks.\n",
```

File: python/docs/source/getting_started/testing_pyspark.ipynb [hunk 4]
```diff
@@ -193,7 +193,7 @@
     "### Option 2: Using [Unit Test](https://docs.python.org/3/library/unittest.html)\n",
```
Comment (allisonwang-db) [inline python/docs/source/getting_started/testing_pyspark.ipynb:193]:
```suggestion
    "### Option 2: Using Unit Test\n",
```

File: python/docs/source/getting_started/testing_pyspark.ipynb [hunk 5]
```diff
@@ -273,7 +273,7 @@
    "source": [
     "### Option 3: Using [Pytest](https://docs.pytest.org/en/7.1.x/contents.html)\n",
```
Comment (allisonwang-db) [inline python/docs/source/getting_started/testing_pyspark.ipynb:274]:
```suggestion
    "### Option 3: Using Pytest\n",
```

**Review by allisonwang-db — COMMENTED** [review]
Thanks for the fix!

**Review by HyukjinKwon — APPROVED** [review]

**Review by allisonwang-db — APPROVED** [review]
Awesome!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by panbingkun** [comment]
late LGTM.

**Pull request closed without merging**